src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+     * Negative delays are treated as to zero delays.
+     *
+     * The first trigger received after the start of the last job execution leads to scheduling of
+     * another execution of the job, either after the default delay or when the currently executing
+     * job is finished, whichever comes last. A newly constructed delayed background job can be
+     * assumed to have started its last job infinitely in the past.
+     * @param delay the maximum trigger aggregation delay in milliseconds
+     * @see #triggerExecution()
+     */
+    @Override
+    public synchronized void triggerExecution(long delay) {
+        tryEnqueue(delay);
+    }
+
+    @Override
+    public synchronized void terminate() {
+        switch (state) {
+            case TERMINATED:
+                assert(waitingTickerJob == null) : "having ticker job in TERMINATED state";
+                assert(thread == null) : "having job thread in TERMINATED state";
+                return;
+            case TERMINATING:
+                assert(waitingTickerJob == null) : "having ticker job in TERMINATING state";
+                assert(thread != null) : "TERMINATING state but no thread";
+                return;
+            case IDLE:
+                toTERMINATED();
+                return;
+            case WAITING:
+                toTERMINATED();
+                return;
+            case RUNNING:
+                toTERMINATING();
+                return;
+        }
+    }
+
+    @Override
+    public synchronized boolean isTerminated() {
+        return state == JobState.TERMINATED;
+    }
+
+    @Override
+    public synchronized void waitForTermination(long timeout) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + timeout;
+        while(timeout > 0 && state != JobState.TERMINATED) {
+            wait(timeout);
+            timeout = deadline - System.currentTimeMillis();
+        }
+    }
+
+    /**
+     * Implementation of {@link #triggerExecution(long)}.
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void tryEnqueue(long delay) {
+        if (state == JobState.TERMINATING || state == JobState.TERMINATED) {
+            return;
+        }
+        if (delay < 0) {
+            delay = 0;
+        }
+        long newExecutionTime = System.currentTimeMillis() + delay;
+        if (newExecutionTime < nextExecutionTime) {
+            nextExecutionTime = newExecutionTime;
+            if (state == JobState.RUNNING) {
+                // Will automatically schedule this run when the running job finishes.
+                return;
+            }
+            if (state == JobState.WAITING) {
+                // Best-effort attempt at removing the stale job to be replaced.
+                ticker.removeQueuedJob(waitingTickerJob);
+                // Replace the ticker job in case the above fails, so the stale job will not run.
```

Comment:
It might make sense to explain why this failing is normal. IIRC, its because for example TrivialTicker will not consider a job as "queued" if it has started already, and thus won't remove it?